test/irmin/test_type.ml

```diff
@@ -251,34 +251,38 @@ let test_to_string () =
       Stdlib.infinity;
       Stdlib.nan;
     |]
-    "[-inf,-0,0,2.220446049250313e-16,inf,nan]";
-  test "(unit * int)" T.(pair unit int) ((), 1) "[{},1]";
-  test "unit option{some}" T.(option unit) (Some ()) "{\"some\":{}}";
-  test "int option{none}" T.(option unit) None "null";
+    "[|neg_infinity; -0.; 0.; 2.22044604925e-16; infinity; nan|]";
+  test "(unit * int)" T.(pair unit int) ((), 1) "((), 1)";
+  test "unit option{some}" T.(option unit) (Some ()) "Some ()";
+  test "int option{none}" T.(option unit) None "None";
   test "(int * string * bool)"
     T.(triple int string bool)
-    (1, "foo", true) "[1,\"foo\",1]";
+    (1, "foo", true) "(1, \"foo\", true)";
   test "(string, bool) result{ok}"
     T.(result string bool)
-    (Ok "foo") "{\"ok\":\"foo\"}";
+    (Ok "foo") "Ok (\"foo\")";
   test "(string, bool) result{error}"
     T.(result string bool)
-    (Error false) "{\"error\":0}";
+    (Error false) "Error (false)";
 
   (* Test cases for algebraic combinators *)
   let open Algebraic in
-  test "enum" my_enum_t Alpha "\"Alpha\"";
-  test "variant" my_variant_t (Right [ 1; 2 ]) "{\"Right\":[1,2]}";
+  test "enum" my_enum_t Alpha "Alpha";
+  test "variant" my_variant_t (Right [ 1; 2 ]) "Right ([1; 2])";
   test "recursive variant" my_recursive_variant_t
     (Branch [ Branch [ Leaf 1 ]; Leaf 2 ])
-    "{\"Branch\":[{\"Branch\":[{\"Leaf\":1}]},{\"Leaf\":2}]}";
+    "Branch ([Branch ([Leaf (1)]); Leaf (2)])";
   test "record" my_record_t
     { foo = 2; flag = false; letter = Delta }
-    "{\"foo\":2,\"flag\":0,\"letter\":\"Delta\"}";
+    {|{ "foo" = 2;
+  "flag" = false;
+  "letter" = Delta }|};
 
   test "recursive record" my_recursive_record_t
     { head = 1; tail = Some { head = 2; tail = None } }
-    "{\"head\":1,\"tail\":{\"head\":2}}";
+    {|{ "head" = 1;
```

Comment:
if you want the output of `pp` to be closer of the OCaml syntax you probably don't want double quotes around `"head"`.